### PR TITLE
CONFIG_NET* cleanup

### PR DESCRIPTION
**Pull request by agansari**
According to #12144  there are unused/undefined CONFIG_NET* macros.
The purpose of this pull request is to fix/remove/correct these.

Code owners, please tell me if you find any issues with these changes.

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]
nice. fix your commit title/message.

But besides that it's good to go.

File: drivers/ethernet/eth_dw.c [hunk 1]
```diff
@@ -136,7 +136,7 @@ static void eth_tx_spin_wait(struct eth_runtime *context)
 
 static void eth_tx_data(struct eth_runtime *context, u8_t *data, u16_t len)
 {
-#ifdef CONFIG_NET_DEBUG_L2_ETHERNET
```
Comment (pfalcon) [inline drivers/ethernet/eth_dw.c]: Should be replaced with CONFIG_ETHERNET_LOG_LEVEL >= LOG_LEVEL_DBG

File: samples/net/common/common.cmake [hunk 2]
```diff
@@ -1,14 +1 @@
 # Common routines used in net samples
-
-if(CONFIG_NET_TESTING)
```
Comment (pfalcon) [inline samples/net/common/common.cmake]: Would like to hear comment from @jukkar here. Maybe we shouldn't remove this chunk, but rename CONFIG_NET_TESTING.

**Review by pfalcon — CHANGES_REQUESTED** [review]

File: drivers/ethernet/eth_dw.c [hunk 3]
```diff
@@ -136,7 +136,7 @@ static void eth_tx_spin_wait(struct eth_runtime *context)
...
+#ifdef CONFIG_ETHERNET_LOG_LEVEL >= LOG_LEVEL_DBG
```
Comment (jukkar) [inline drivers/ethernet/eth_dw.c]: Shouldn't this be written as `#if CONFIG_ETHERNET_LOG_LEVEL >= LOG_LEVEL_DBG` (so not with #ifdef)

File: samples/net/common/common.cmake (hunk 2 above)
Comment (jukkar) [reply samples/net/common/common.cmake]: I managed to find the commit in 2016 that removed this from Kconfig. It really looks like we do not need this setting any more.

**Review by jukkar — DISMISSED** [review]

File: drivers/ethernet/eth_dw.c (hunk 3 above)
Comment (agansari) [reply drivers/ethernet/eth_dw.c]: you are right redone it

**Review by pfalcon — APPROVED** [review]

**Comment by agansari** [comment]
@jukkar @galak made the changes you requested

**Review by jukkar — APPROVED** [review]

**Pull request merged**